src/pages/Applications/Filter/DateInput.tsx

```diff
@@ -0,0 +1,29 @@
+import { ErrorMessage } from "@hookform/error-message";
+import { FieldValues, Path, useFormContext } from "react-hook-form";
+import { FormValues } from "./types";
+
+export default function DateInput<T extends FieldValues>({
+  name,
+}: {
+  name: Path<T>;
+}) {
+  const {
+    register,
+    formState: { errors },
+  } = useFormContext<FormValues>();
+  return (
+    <div>
+      <input
+        {...register(name as any)}
```

Comment:
```suggestion
        {...register(name)}
```
See comment about using `useFormContext<T>();`

Reply:
fixed